kyaml/kio/byteio_reader_test.go

```diff
@@ -372,6 +372,27 @@ metadata:
 			},
 			instance: ByteReader{},
 		},
+
+		//
+		//
+		//
+		{
+			name: "any_content_after_document_separator_should_be_ignored",
+			input: `
+a: b
+--- foo
```

Comment:
would it make more sense to throw an error in this case or just discard it like we would a comment?

let's also add a test with a comment, `--- #foo`, as a regression test

@KnVerey

Reply:
I can add a couple more tests (one for whitespace and another for comments).

As far as detecting invalid content after `---` I chose not to do that in this PR because that adds quite a bit more complexity.  Instead of just splitting, it would have to find and detect the invalid situations as well.  I can do it but was trying to keep this PR as simple as possible.

---

Also, interestingly, I just now discovered that k/k doesn't handle the following as I thought it would:
```
kubectl --dry-run=server apply -f - << EOF
apiVersion: v1
kind: Namespace
metadata:
  name: foo
--- # something
apiVersion: v1
kind: Namespace
metadata:
  name: bar
EOF
```

It doesn't fail, but it only creates foo, not bar.

Output:
```
namespace/foo created (server dry run)
```

It does seem to allow whitespace after `---` though.

Reply:
> Instead of just splitting, it would have to find and detect the invalid situations as well. I can do it but was trying to keep this PR as simple as possible.

That makes sense we can keep it simple. Could you add a TODO comment, file a followup issue, and link the issue in the TODO? 

> It doesn't fail, but it only creates foo, not bar.

Like I said on the issue, that sounds like a bug they need to fix. It's good that we are trying to address it in kustomize.

Reply:
And thanks for the extra tests, they look great!

Reply:
@natasha41575 I just pushed another commit with a version that detects anything other than white space and comments and returns an error.

My original intent was to go for a simple change, but if you are interested in it being more robust in checking the separator lines, take a look at what I just pushed.

Sorry for getting the communication lines crossed up.  Please take a look and if you want to opt for the original, simple version I can revert back to that. :smile:

Reply:
awesome, thank you! Yes I prefer robust checking so the error check is great.